### PR TITLE
Alf setup pr

### PR DESCRIPTION
**Pull request by ymuwakki**
container to setup alfresco using a python script

### CODE REVIEW

File: artifacts/setup_alfresco.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+# Case Files,Complaints,Document Repositories,Expenses,People,Recycle Bin,Tasks,Timesheets,User Profile,Business Processes,Consultations,SAR,Requests
+# ACM
+# Case Files,Complaints,Document Repositories,Requests,Tasks,Consultations,SAR
+
+content_services_path = '/alfresco/api/-default-/public/alfresco/versions/1'
+ags_services_path = '/alfresco/api/-default-/public/gs/versions/1'
+filename = './alfresco_setup.properties'
+
+
+def remove_eol(string: str):
```
Comment (fabricetriboix) [inline artifacts/setup_alfresco.py]: You can use `str.strip("\n")` instead of re-writing it yourself

File: artifacts/alfresco_setup.properties [hunk 2]
```diff
@@ -0,0 +1,8 @@
+http://ec2-54-224-31-118.compute-1.amazonaws.com:8080,admin,admin
+yasser
```
Comment (fabricetriboix) [inline artifacts/alfresco_setup.properties]: The values in this file seem to very "Yasser-specific"... Can you put some more generic values?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      IMAGE_SOURCE=https://github.com/ArkCase/ark_alfresco_setup \
+      MAINTAINER="Armedia LLC"
+
+WORKDIR /app
+
+# Set default docker_context.
+ARG resource_path=artifacts
+
+COPY ${resource_path}/setup_alfresco.py .
+COPY ${resource_path}/alfresco_setup.properties .
```
Comment (fabricetriboix) [inline Dockerfile]: There is a problem here: The Alfresco configuration to apply is embedded inside the container. That means that each time you want to have a different configuration (even a very small change), you will need to build a new container. I think you should design this alfresco_setup container such that it expects the properties file to be mounted inside the container. 

When running in Kubernetes/Helm, this will most probably take the form of a ConfigMap. So the Alfresco setup will truly be a input to the Helm chart and can easily be tuned for each deployment.

**Review by fabricetriboix — DISMISSED** [review]

File: Dockerfile (hunk 3 above)
Comment (ymuwakki) [reply Dockerfile]: Good point.  We decided to convert it into 9 environment variables so no more properties file

File: artifacts/setup_alfresco.py (hunk 1 above)
Comment (ymuwakki) [reply artifacts/setup_alfresco.py]: Using environment variables so the line feeds are no longer there

**Review by naveen-armedia — COMMENTED** [review]
@ymuwakki  Is this going to be an init container for Arkcase application..?

File: artifacts/alfresco_setup.properties [hunk 4]
```diff
@@ -0,0 +1,8 @@
+https://alfresco.armcsbs.org,admin,admin
```
Comment (naveen-armedia) [inline artifacts/alfresco_setup.properties]: @ymuwakki could we leverage ConfigMap in kubernetes for this properties file. So all these values inside property file could be given deploy stage.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,31 @@
+#  Alfresco Setup build
+
+#FROM 345280441424.dkr.ecr.ap-south-1.amazonaws.com/ark_base:latest
+FROM centos:7
```
Comment (fabricetriboix) [inline Dockerfile]: Why did you change from our base image to the official centos image?

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      IMAGE_SOURCE=https://github.com/ArkCase/ark_alfresco_setup \
+      MAINTAINER="Armedia LLC"
+
+WORKDIR /app
+
+# Set default docker_context.
+ARG resource_path=artifacts
+
+COPY ${resource_path}/setup_alfresco.py .
+COPY ${resource_path}/alfresco_setup.properties .
```
Comment (fabricetriboix) [inline Dockerfile]: I thought you didn't need this properties file anymore because you are now using environment variables?

File: artifacts/alfresco_setup.properties (hunk 4 above)
Comment (fabricetriboix) [reply artifacts/alfresco_setup.properties]: @ymuwakki I thought you said that you are now using environment variables, so shouldn't this file go away?

Comment (ymuwakki) [reply artifacts/alfresco_setup.properties]: yes. removed.

File: Dockerfile (hunk 6 above)
Comment (ymuwakki) [reply Dockerfile]: yes. removed.

**Review by naveen-armedia — APPROVED** [review]

**Review by fabricetriboix — APPROVED** [review]
